pkgs/development/compilers/llvm/13/default.nix

```diff
@@ -236,7 +236,7 @@ in let
         # https://github.com/NixOS/nixpkgs/pull/194634#issuecomment-1272129132
         ++ lib.optional (
           stdenv.targetPlatform.isDarwin
-            && !stdenv.targetPlatform.isAarch64
+            && darwin.apple_sdk ? sdk
```

Comment:
The LLVM checks can be replaced with `stdenv.targetPlatform.isDarwin && lib.versionOlder stdenv.targetPlatform.darwinSdkVersion "11.0"`. Even when using `overrideSDK` and after the refactor, that version will always match the version of the SDK in use.

Reply:
most of the llvm checks, like this one, also check for `(lib.versionOlder darwin.apple_sdk.sdk.version "11.0")` which I assume can be dropped when using `lib.versionOlder stdenv.targetPlatform.darwinSdkVersion "11.0"`?

Reply:
Also for what it’s worth, I’m pretty sure this will compile on the 10.12 SDK after the ld64 update. Libsystem pulls the affected header from ld64, which will be considerably newer than the 10.12 SDK. I actually consider that a bug, but there’s no nice way to fix it right now.

Reply:
> most of the llvm checks, like this one, also check for `(lib.versionOlder darwin.apple_sdk.sdk.version "11.0")` which I assume can be dropped when using `lib.versionOlder stdenv.targetPlatform.darwinSdkVersion "11.0"`?

Right. I only commented on this one, but all the identical checks could be replaced in the same way.

Reply:
thanks -- i miss-read.